pkg/actuators/machine/instances.go

```diff
@@ -374,3 +375,27 @@ func (il instanceList) Less(i, j int) bool {
 func sortInstances(instances []*ec2.Instance) {
 	sort.Sort(instanceList(instances))
 }
+
+func getInstanceMarketOptionsRequest(providerConfig *awsproviderv1.AWSMachineProviderConfig) *ec2.InstanceMarketOptionsRequest {
+	if providerConfig.SpotMarketOptions == nil {
+		// Instance is not a Spot instance
+		return nil
+	}
+
+	// Set required values for Spot instances
+	spotOptions := &ec2.SpotMarketOptions{}
+	spotOptions.SetInstanceInterruptionBehavior(ec2.InstanceInterruptionBehaviorTerminate)
```

Comment:
can we add a clarifying comment here? e.g we are intentionally gating here to enforce mapi lifecycle design invariants machine 1 - 1 instance